test/miscellaneous_tests/api.jl

```diff
@@ -446,3 +446,56 @@ let
     neweqs = getfield.(equations(ns), :rhs)
     @test_throws MethodError Catalyst.to_multivariate_poly(neweqs)
 end
+
```

Comment:
Add a test with a registered function that takes `t` as a parameter within the rate to make sure that is caught as non-autonomous?

Reply:
added